SwiftVideoBackground/BackgroundVideo.swift

```diff
@@ -6,87 +6,77 @@
 //  Copyright © 2016 Wilson Ding. All rights reserved.
 //
 
-import UIKit
 import AVFoundation
+import UIKit
 
-public class BackgroundVideo: UIView {
-
-    /// isMuted - provides the ability to set whether background video is muted
-    public var isMuted = false
-
-    private var player: AVPlayer?
-
-    /**
-     createBackgroundVideo(name: String, type: String) function:
-     - name: String - take in the name of the video file
-     - type: String - take in the file type of the video file
-     */
-
-    public func createBackgroundVideo(name: String, type: String) {
-        do {
-            try createBackground(name: name, type: type)
-        } catch {
-            print(error.localizedDescription)
+/// Class that plays a video on a UIView.
+public class BackgroundVideo {
+    private var player = AVPlayer()
+
+    /// Plays a video on a given UIView.
+    ///
+    /// - Parameters:
+    ///     - view: UIView that the video will be played on.
+    ///     - videoName: String name of video that you have added to your project.
+    ///     - videoType: String type of the video. e.g. "mp4"
+    ///     - isMuted: Bool indicating whether video is muted. Defaults to true.
+    ///     - opacity: The opacity of the layer, as a value between zero and one.
+    ///                Defaults to one. Specifying a value outside the [0,1]
+    ///                range will give undefined results.
+    ///     - loopVideo: Bool indicating whether video should restart when finished.
+    ///                  Defaults to true.
+    public func play(view: UIView,
+                     videoName: String,
+                     videoType: String,
+                     isMuted: Bool = true,
+                     opacity: Float = 1.0,
+                     loopVideo: Bool = true) {
+        guard let path = Bundle.main.path(forResource: videoName, ofType: videoType) else {
+            print("BackgroundVideo: [ERROR] could not find video")
+            return
         }
-    }
-
-    /**
-      createBackgroundVideo(name: String, type: String, alpha: CGFloat) function:
-      - name: String - take in the name of the video file
-      - type: String - take in the file type of the video file
-      - alpha: CGFloat - take in the desired alpha/darkness of the background video
-    */
 
-    public func createBackgroundVideo(name: String, type: String, alpha: CGFloat) {
-        createAlpha(alpha: alpha)
-
-        do {
-            try createBackground(name: name, type: type)
-        } catch {
-            print(error.localizedDescription)
+        let url = URL(fileURLWithPath: path)
+        player = AVPlayer(url: url)
+        player.isMuted = isMuted
+        player.play()
+
+        let layer = AVPlayerLayer(player: player)
+        layer.frame = view.frame
+        layer.opacity = opacity
```

Comment:
Previous versions overlaid a solid black subview over the player UIView, where `opacity` would adjust the alpha/transparency of that black subview (essentially "darkening" the background video). This helped objects in the foreground such as labels/buttons pop out.

Do we want to shift to changing the transparency of the player itself? (I do see potential benefit of having one less subview, but worry about potentially having the video look "faded".)

Reply:
are you saying the old way makes the videos look better? if so i'll change back to that way. the only reason why i'm doing it this way in the PR is because it was the first solution I came across :D

Reply:
Haha I think either solution (darkening the video, or making the video transparent) should be ok.. Maybe we should offer both and let the user of the library decide what they want? Although I feel offering multiple ways to do something might unnecessarily complicate the library.

Reply:
It seems like the current way could potentially look better at a minor if any cost. I'll change to back to the current way.

Yea we can just keep it simple for now.